zeebe/logstreams/src/main/java/io/camunda/zeebe/logstreams/impl/flowcontrol/RateMeasurement.java

```diff
@@ -34,6 +34,15 @@ public RateMeasurement(
     observations = new LinkedBlockingDeque<>(numberOfObservations);
   }
 
+  /**
+   * Updates the rate measurement with the given value.
+   *
+   * <p>The new observation must be the first or measured after the previous at least by the
+   * resolution time.
```

Comment:
:x: That's not correct, the observation can be made at any time. It's just that we ignore it in that case.